packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts

```diff
@@ -0,0 +1,56 @@
+import { IDropdownOption, IDropdownProps } from '../../Dropdown';
+import { IIconProps } from '../../Icon';
+
+export interface IComboBox {
+
+}
+
+export interface IComboBoxProps extends IDropdownProps {
+
+  /**
+   * Collection of options for this ComboBox
+   */
+  options?: IComboBoxOption[];
+
+  /**
+   * Callback issues when either:
+   * 1) the selected option changes
+   * 2) a manually edited value is submitted. In this case there may not be a matched option if allowFreeform is also true
+   *    (and hence only value would be true, the other parameter would be null in this case)
+   */
+  onChanged?: (option?: IComboBoxOption, index?: number, value?: string) => void;
+
+  /**
+   * Callback issued when the options should be resolved, if they have been updated or
+   * if they need to be passed in the first time
+   */
+  onResolveOptions?: () => IComboBoxOption[] | PromiseLike<IComboBoxOption[]>;

```

Comment:
One is handling the full state outside of the component (e.g. setting the options directly) opposed to the uncontrolled case where it may take some time to get the options back but we want the state to be stored inside of the component. The promise allows for the update without blocking